compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -118,8 +122,8 @@ impl<'a> Drop for Parser<'a> {
 struct TokenCursor {
     frame: TokenCursorFrame,
     stack: Vec<TokenCursorFrame>,
-    cur_token: Option<TreeAndSpacing>,
-    collecting: Option<Collecting>,
+    num_next_calls: usize,
+    num_next_desugared_calls: usize,
```

Comment:
Is it necessary to keep both `num_next_calls` and `num_next_desugared_calls`?
`desugar_doc_comments` is an immutable configuration property of a `Parser`, so any given parser will either always increment `num_next_calls`s or always increment `num_next_desugared_calls`.